### PR TITLE
ARROW-17741: [Packaging] Include JDBC driver in java-jars artifacts

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

**Comment by lidavidm** [comment]
@github-actions crossbow submit java-jars

**Comment by davisusanibar** [comment]
New jar available at: https://github.com/ursacomputing/crossbow/releases/tag/actions-b9e5302f5d-github-java-jars

File: dev/tasks/tasks.yml [hunk 1]
```diff
@@ -927,6 +927,11 @@ tasks:
       - flight-sql-{no_rc_version}-tests.jar
       - flight-sql-{no_rc_version}.jar
       - flight-sql-{no_rc_version}.pom
+      - flight-sql-jdbc-driver-{no_rc_version}-javadoc.jar
+      - flight-sql-jdbc-driver-{no_rc_version}-sources.jar
+      - flight-sql-jdbc-driver-{no_rc_version}-tests.jar
+      - flight-sql-jdbc-driver-{no_rc_version}.jar
+      - flight-sql-jdbc-driver-{no_rc_version}.pom
```
Comment (raulcd) [inline dev/tasks/tasks.yml:934]: As this was merged, can we update it to: https://github.com/apache/arrow/commit/bced07de986b9f7baaa4885d7f03f4e46d1abed8
```suggestion
      - flight-sql-jdbc-driver-{no_rc_snapshot_version}-javadoc.jar
      - flight-sql-jdbc-driver-{no_rc_snapshot_version}-sources.jar
      - flight-sql-jdbc-driver-{no_rc_snapshot_version}-tests.jar
      - flight-sql-jdbc-driver-{no_rc_snapshot_version}.jar
      - flight-sql-jdbc-driver-{no_rc_snapshot_version}.pom
```

**Comment by lidavidm** [comment]
@github-actions crossbow submit java-jars

**Review by raulcd — APPROVED** [review]
Thanks! LGTM

**Comment by raulcd** [comment]
on the java-jars job the new artifacts:
```
                    flight-sql-jdbc-driver-10.0.0-SNAPSHOT-javadoc.jar [     OK]
                   flight-sql-jdbc-driver-10.0.0-SNAPSHOT-sources.jar [     OK]
                     flight-sql-jdbc-driver-10.0.0-SNAPSHOT-tests.jar [     OK]
                           flight-sql-jdbc-driver-10.0.0-SNAPSHOT.jar [     OK]
                           flight-sql-jdbc-driver-10.0.0-SNAPSHOT.pom [     OK]
```

**Pull request merged**